rfd/0XYZ-nested-accesslists.md

```diff
@@ -0,0 +1,121 @@
+---
+title: RFD 0XYZ - Nested Access lists
+authors: Alex McGrath (alex.mcgrath@goteleport.com)
+state: in development
+---
+
+# Related
+  * Partial implementation PR: https://github.com/gravitational/teleport/pull/38738
+
+# What
+
+This RFD proposes adding support for including owners and
+members in access lists from other access lists
+
+# Why
+
+This allows a hierarchical access list structure for organizations
+which is supported by access list implementations being integrated
+with, for example Azure Entra supports including groups in other
+groups which we want to be able to mimic the structure of in Teleport
+access lists.
+
+Users in an access list hierarchy will inherit the granted roles and
+traits for members and owners from access lists referencing the lists
+they're in.
+
+# Implementations
+
+New fields will be introduced into the access_list type:
+
+```yaml
+kind: access_list
+metadata:
+  name: e69aa529-2a7f-4de2-87be-97fd94309a9f
+spec:
+  grants:
+    roles:
+    - access
+    - auditor
+    traits: {}
+  # list of references to other access lists, for users to include in this access list
+  member_access_lists:
+  - name: ea4cbbc7-bee1-49b3-bf78-734b4b27ea38
+  # list of references to other access lists, for owners to include in this access list
+  owner_access_lists:
+  - name: 3e9df1e7-0b8a-4984-b2e8-5bc0d7b356a9
+  title: access-list-a
+version: v1
+```
+
+These fields will contain a list of other access lists that will be
+included in the access list.
+
+# Implementation considerations
+
+The implementation will not support looping within the heirarchy as
```

Comment:
Probably would be helpful to mention if there would be any maximum depth we would support. Which I think we do need to have and set it to something reasonable, like 10? We can always extend it later if needed.